core/src/main/java/org/elasticsearch/index/mapper/DocumentMapper.java

```diff
@@ -128,18 +128,8 @@ public Builder put(MetadataFieldMapper.Builder<?, ?> mapper) {
             return this;
         }
 
-        public Builder transform(ScriptService scriptService, Script script) {
-            sourceTransforms.add(new ScriptTransform(scriptService, script));
-            return this;
-        }
-
-        /**
-         * @deprecated Use {@link #transform(ScriptService, Script)} instead.
-         */
-        @Deprecated
-        public Builder transform(ScriptService scriptService, String script, ScriptType scriptType, String language,
-                Map<String, Object> parameters) {
-            sourceTransforms.add(new ScriptTransform(scriptService, new Script(script, scriptType, language, parameters)));
+        public Builder transform(SourceTransform sourceTransform) {
```

Comment:
This is the other problem I have. Transforms aren't really pulled out of mappings, the implementation is just moved out (which makes this *more* configurable and harder to remove).